data/account/participationRegistry.go

```diff
@@ -0,0 +1,108 @@
+// Copyright (C) 2019-2021 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package account
+
+import (
+	"errors"
+	"github.com/algorand/go-algorand/crypto"
+	"github.com/algorand/go-algorand/data/basics"
+	"github.com/algorand/go-algorand/util/db"
+)
+
+// ParticipationID identifies a particular set of participation keys.
+type ParticipationID crypto.Digest
+
+// ParticipationRecord contains all metadata relating to a set of participation keys.
+type ParticipationRecord struct {
+	ParticipationID ParticipationID
+
+	Account     basics.Address
+	FirstValid  basics.Round
+	LastValid   basics.Round
+	KeyDilution uint64
+
+	LastVote               basics.Round
+	LastBlockProposal      basics.Round
+	LastCompactCertificate basics.Round
+	EffectiveFirst         basics.Round
+	EffectiveLast          basics.Round
+
+	// VRFSecrets
+	// OneTimeSignatureSecrets
+}
+
+// ParticipationAction is used when recording participation actions.
+type ParticipationAction int
+
+// ParticipationAction types
+const (
+	Vote ParticipationAction = iota
+	BlockProposal
+	CompactCertificate
+)
+
+// ParticipationIDNotFoundErr is used when attempting to update a set of keys which do not exist.
+var ParticipationIDNotFoundErr error
```

Comment:
nit: errors should start with ErrXXXX.
also - defining the error as 
```golang
var ParticipationIDNotFoundErr = errors.New("the participation ID was not found")
```
is perfectly fine.. no need to overkill it with init() functions.